src/matching/Match_patterns.ml

```diff
@@ -139,7 +139,8 @@ let (rule_id_of_mini_rule : Mini_rule.t -> Pattern_match.rule_id) =
     langs = mr.langs;
   }
 
-let match_rules_and_recurse m_env (file, hook, matches) rules matcher k any x =
+let match_rules_and_recurse m_env (file, source, hook, matches) rules matcher k
```

Comment:
A 4-tuple is getting a bit ugly, should we make this a record?

Reply:
I agree, this is not super clean. I was looking for a smallish diff here, but just making this a record wouldn't be too bad.

Reply:
I don't mind if the diff gets big after the tricky parts have been resolved.

Reply:
Done in bdb51b197b7ee4fc901ad1f3d34313f2e25b10f0